balena/models/hup.py

```diff
@@ -0,0 +1,238 @@
+import semver
+
+from .. import exceptions
+
+
+class Hup(object):
```

Comment:
@nghiant2710 why is this part really needed? I would rather not hardcode the device types here as well, as we are trying to remove in other tools too. The proxy already gatekeeping the allowed versions, and in the NodeSDK it's just a stopgap measure to remove these hardcoded values. 

In generally also IMHO do not concern about 1.x-1.x and 1.x-2.x updates as they are going to be deprecated soon, and the default for 2.x-2.x updates is that it should be allowed. 

Thus:

* device type needs not be checked
* updates for 2.x devices should be allowed in general
* the dev, downgrade, same version checks are good, and can stay
* The min target version for 2.x can be just 2.2.0+rev1 in general.

These checks are in the node SDK's modules, I understand but they are getting cleaned from there. The proxy does its checks as well, no need to repeat everything like this, IMHO we can let the tests be in the proxy alone (where it will be generally more permissive, as to no need to add new device types explicitly to add support for them in HUP, that's work in progress)

This part above probably mainly concerns `get_hup_action_type` further below in the module, I guess, but some parts this whole file.

Let me know if anything's strange about this mentioned above!

Reply:
Got it, I don't know much about how the hup proxy works so just tried to include every single check we have in the JS SDK. Updated!